### PR TITLE
Rosetta proposal internal review

### PR DESCRIPTION
**Pull request by psy2848048**

### CODE REVIEW

File: applications/rosetta_integration.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+- Build collector, parser on EOSIO's txs and store our dedicated RDB with keeping txs' history
+  - Many API support historical data lookup, then offchain RDB processing is inevitable.
+  - Please check our similar work. (Stated in Additional information)
+  - We can cover both available in gRPC or REST connection (prefer gRPC)
+- Data source
+  - Build from public snapshots of the data depot
+  - Use public endpoint of BPs'
+- Implement Rosetta server following its API specification
+  - Language: Golang on [`rosetta-sdk-go`](https://github.com/coinbase/rosetta-sdk-go)
+- Maintain the intrastructure for operation. (Publicize our API server & endpoint)
```
Comment (jbamlee) [inline applications/rosetta_integration.md]:
```suggestion
- Maintain the infrastructure for operation. (Publicize our API server & endpoint)
```

File: applications/rosetta_integration.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+- Documentation & developer support
+
+### Ecosystem Fit
+
+- Make lower barrier to developers
+  - Who: Who don't have to know the knowledge of Antelope but needs some information of the Antelope (e.g. )
+  - How: Provide the APIs which they already know and have integrated before. Don't have to learn Antelope-specific API.
+  - Where would be good: Exchange developer, Blockchain statisics application (CoinGecko, CMC, etc), Multichain bridging project, etc
```
Comment (jbamlee) [inline applications/rosetta_integration.md]:
```suggestion
  - Where would be good: Exchange developer, Blockchain statistics application (CoinGecko, CMC, etc), Multichain bridging project, etc
```

File: applications/rosetta_integration.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+- Young LEE: [https://www.linkedin.com/in/jihyunglee/](https://www.linkedin.com/in/jihyunglee/)
+- Derick MOON: [https://www.linkedin.com/in/sanghoon-moon-79b5a81b2/](https://www.linkedin.com/in/sanghoon-moon-79b5a81b2/)
+- Sooyoung HA: [https://www.linkedin.com/in/sooyoung-ha-9a0195140/](https://www.linkedin.com/in/sooyoung-ha-9a0195140/)
+- Bryan RHEE: [https://www.linkedin.com/in/psy2848048/](https://www.linkedin.com/in/psy2848048/)
+- Joowon YUN: [https://www.linkedin.com/in/joowon-yun-520a1557/](https://www.linkedin.com/in/joowon-yun-520a1557/)
+- Maro KIM: [https://www.linkedin.com/in/maro/](https://www.linkedin.com/in/maro/)
+
+## Development Status
+
+7 developers (1 FE, 1 data, 5 BE & system) and 1 product designer are working in DELIGHT LABS. We had fully experience in Cosmos SDK projects - both of validation & dApp building. Especially, we validate 10+ projects including Terra classic, Terra 2.0, ORBS, Xpla, Oasis protocol, Fetch.ai, Findora, Casperlabs, SSV, Rizon, Medibloc, and so on, and we build & operate 1st dApp and AMM DEX - Terraswap on both of Terra classic & 2.0.
```
Comment (jbamlee) [inline applications/rosetta_integration.md]:
```suggestion
7 developers (1 FE, 1 data, 5 BE & system) and 1 product designer are working in DELIGHT LABS. We had full experience in Cosmos SDK projects - both of validation & dApp building. Especially, we validate 10+ projects including Terra classic, Terra 2.0, ORBS, Xpla, Oasis protocol, Fetch.ai, Findora, Casperlabs, SSV, Rizon, Medibloc, and so on, and we build & operate 1st dApp and AMM DEX - Terraswap on both of Terra classic & 2.0.
```

File: applications/rosetta_integration.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+- **Estimated duration:** 8 weeks
+- **FTE:** 1
+- **Costs:** USD 25K
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License |  MIT  |
+| 0b. | Documentation | Module level documentation & README outline |
+| 0c. | CI test | No working unittest but make the code be ready on Makefile & Github action |
```
Comment (jbamlee) [inline applications/rosetta_integration.md]:
```suggestion
| 0c. | CI test | No working unit test but make the code be ready on Makefile & Github action |
```

File: applications/rosetta_integration.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+### Milestone 2 - Data API & Contruction API
+
+#### Action items
+
+- Implement Data API & Contruction API without offchain data import
```
Comment (jbamlee) [inline applications/rosetta_integration.md]:
```suggestion
- Implement Data API & Construction API without off-chain data import
```

File: applications/rosetta_integration.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+| 0b. | API server | HTTPS wrapped API server. The endpoints stated above should work |
+| 0c. | Code | Ready-to-build state |
+| 0d. | CI test | Unit test coverage >= 70% |
+
+### Milestone 3 - Indexer API
+
+#### Action items
+
+- Subscribe data from the data depot & reorganize in the dedicated RDB
+- Apply aggregator for histrical lookup support
```
Comment (jbamlee) [inline applications/rosetta_integration.md]:
```suggestion
- Apply aggregator for historical lookup support
```

File: applications/rosetta_integration.md [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+| 0a. | License |  MIT  |
+| 0b. | API server | HTTPS wrapped API server. Historical lookup is also supported |
+| 0c. | Code | Ready-to-build state |
+| 0d. | CI test | Unit test coverage >= 70% |
+
+### Milestone 4 - Documentation & its development
+
+#### Action items
+
+- Deatailing MD documentation and improve its quality
```
Comment (jbamlee) [inline applications/rosetta_integration.md]:
```suggestion
- Detailing MD documentation and improving its quality
```

File: applications/rosetta_integration.md [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+## Future Plans
+
+- Once the API server is deployed, we will operate at least a year with code upgrade. Moreover, we will support bugfix with enough years.
+- Currently very few EOSIO-based tokens are listed on CEXes. But we wish that many Antelope-based token would be easily listed, easy like ERC20. Not only noetwork token like EOS, but also other tokens would be supported equally on `/account/balance`.
+- We will also influence our works and give many helps to relayers from/to Antelope to check Tx state tracking.
+
+## Additional Information
+
+- Our similar reference - [Cosmwasm-ETL](https://github.com/DELIGHT-LABS/cosmwasm-etl) (private, building)
+  - For unifying data processing codes to cover all DELIGH LABS's DEXes built on Cosmwasm-based projects
```
Comment (jbamlee) [inline applications/rosetta_integration.md]:
```suggestion
  - For unifying data processing codes to cover all DELIGHT LABS' DEXes built on Cosmwasm-based projects
```

File: applications/rosetta_integration.md [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | API server | HTTPS wrapped API server. Historical lookup is also supported |
+| 0c. | Audit report | Audit from reliable orgs. Get a `safe` remark |
+
+## Future Plans
+
+- Once the API server is deployed, we will operate at least a year with code upgrade. Moreover, we will support bugfix with enough years.
+- Currently very few EOSIO-based tokens are listed on CEXes. But we wish that many Antelope-based token would be easily listed, easy like ERC20. Not only noetwork token like EOS, but also other tokens would be supported equally on `/account/balance`.
```
Comment (jbamlee) [inline applications/rosetta_integration.md]:
```suggestion
- Currently, very few EOSIO-based tokens are listed on CEXes. But we wish that many Antelope-based tokens would be easily listed, easy like ERC20. Not only network token like EOS, but also other tokens would be supported equally on `/account/balance`.
```

**Review by jbamlee — APPROVED** [review]

File: applications/rosetta_integration.md (hunk 9 above)
Comment (psy2848048) [reply applications/rosetta_integration.md]: https://github.com/DELIGHT-LABS/grant-framework/pull/1/commits/d33af06919335fb421c3e51922cb6af2a31604f8 🙇‍♂️

File: applications/rosetta_integration.md (hunk 8 above)
Comment (psy2848048) [reply applications/rosetta_integration.md]: https://github.com/DELIGHT-LABS/grant-framework/pull/1/commits/d33af06919335fb421c3e51922cb6af2a31604f8 🙇‍♂️

File: applications/rosetta_integration.md (hunk 7 above)
Comment (psy2848048) [reply applications/rosetta_integration.md]: https://github.com/DELIGHT-LABS/grant-framework/pull/1/commits/d33af06919335fb421c3e51922cb6af2a31604f8 🙇‍♂️

File: applications/rosetta_integration.md (hunk 6 above)
Comment (psy2848048) [reply applications/rosetta_integration.md]: https://github.com/DELIGHT-LABS/grant-framework/pull/1/commits/d33af06919335fb421c3e51922cb6af2a31604f8 🙇‍♂️

File: applications/rosetta_integration.md (hunk 5 above)
Comment (psy2848048) [reply applications/rosetta_integration.md]: https://github.com/DELIGHT-LABS/grant-framework/pull/1/commits/d33af06919335fb421c3e51922cb6af2a31604f8 🙇‍♂️

File: applications/rosetta_integration.md (hunk 4 above)
Comment (psy2848048) [reply applications/rosetta_integration.md]: https://github.com/DELIGHT-LABS/grant-framework/pull/1/commits/d33af06919335fb421c3e51922cb6af2a31604f8 🙇‍♂️

File: applications/rosetta_integration.md (hunk 3 above)
Comment (psy2848048) [reply applications/rosetta_integration.md]: https://github.com/DELIGHT-LABS/grant-framework/pull/1/commits/d33af06919335fb421c3e51922cb6af2a31604f8 🙇‍♂️

File: applications/rosetta_integration.md (hunk 2 above)
Comment (psy2848048) [reply applications/rosetta_integration.md]: https://github.com/DELIGHT-LABS/grant-framework/pull/1/commits/d33af06919335fb421c3e51922cb6af2a31604f8 🙇‍♂️

File: applications/rosetta_integration.md (hunk 1 above)
Comment (psy2848048) [reply applications/rosetta_integration.md]: https://github.com/DELIGHT-LABS/grant-framework/pull/1/commits/d33af06919335fb421c3e51922cb6af2a31604f8 🙇‍♂️

**Pull request merged**